src/OidcClientSettings.ts

```diff
@@ -137,6 +137,12 @@ export interface OidcClientSettings {
      * Only scopes in this list will be passed in the token refresh request.
      */
     refreshTokenAllowedScope?: string | undefined;
+
+    /**
+     * https://datatracker.ietf.org/doc/html/rfc6749#section-3.3 describes behavior when omitting scopes from sign in requests
+     * If the IDP supports default scopes, this setting will ignore the scopes property passed to the config
+     */
+    useIdpDefaultScopes?: boolean;
```

Comment:
the current behavior when not specifying a scope value is to default the value to "openid". Changing that would be a breaking change for the library. 

This field was added to prevent a breaking change.  

If the maintainers are OK with a breaking change, I think it would be more straightforward to remove the default value for scope instead of adding this new property. I can set up a PR for that approach if it is desired over adding this property.

Reply:
Good point, lets keep this settings property for now. This give us time to decide when to do the breaking change. I will have to make a v4.0.0 for this change...

Can you immediately deprecate the new property?

Reply:
sure thing.  The deprecated tag has been added and pushed. Thanks @pamapa 

I was looking more into this implementation. I think it would be awesome if the library supported this feature, but I realized that the doc I linked to in the description is the part of the [OAuth spec file](https://datatracker.ietf.org/doc/html/rfc6749#section-3.3). The [OIDC authentication request spec](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest) does state that the `openid` scope is required for a valid authentication request.

The OIDC specs  does also state
> "If the openid scope value is not present, the behavior is entirely unspecified." 

If it would make it more clear, I could also rename the property to `allowOidcScopeOmission` or something similar.

Reply:
> The [OIDC authentication request spec](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest) does state that the `openid` scope is required for a valid authentication request.

Oh yes, that makes the `@deprecated` invalid :-) Also its better to keep the type of scope `string`. I guess you can deal with empty string...

According to https://datatracker.ietf.org/doc/html/rfc6749#section-3.3:
"If the client omits the scope parameter when requesting
   authorization, the authorization server MUST either process the
   request using a pre-defined default value or fail the request
   indicating an invalid scope.  The authorization server SHOULD
   document its scope requirements and default value (if defined)."
the new property could be named `omitScopeWhenRequesting`?

What kind of Idp are you using for this feature?

Reply:
We're using ForgeRock (now part of Ping Identity) as our IDP. We do have the ability to define the IDP behavior when scope is not provided. 

I've removed the updates to the scope typing so it is no longer optional where it was not optional before. The new property has been renamed `omitScopeWhenRequesting` and that property value is solely responsible for including the `scope` query param in the authorization requests now.   The overall changes in this PR are much smaller now. Thanks for all the feedback and suggestions.